sdk/src/port/out/hs/walletconnect/HederaWalletConnectTransactionAdapter.ts

```diff
@@ -0,0 +1,286 @@
+import Account from '../../../../domain/context/account/Account';
+import TransactionResponse from '../../../../domain/context/transaction/TransactionResponse.js';
+import { Environment } from '../../../../domain/context/network/Environment';
+import { SupportedWallets } from '../../../../domain/context/network/Wallet';
+
+@singleton()
+/**
+ * Represents a transaction adapter for Hedera Wallet Connect.
+ */
+export class HederaWalletConnectTransactionAdapter extends HederaTransactionAdapter {
+	public account: Account;
+	protected network: Environment;
+	protected dAppConnector: DAppConnector | undefined;
+
+	constructor(
+		@lazyInject(EventService)
+		public readonly eventService: EventService,
+		@lazyInject(NetworkService)
+		public readonly networkService: NetworkService,
+		@lazyInject(MirrorNodeAdapter)
+		public readonly mirrorNodeAdapter: MirrorNodeAdapter,
+		@lazyInject(QueryBus)
+		public readonly queryBus: QueryBus,
+	) {
+		super(mirrorNodeAdapter, networkService);
+	}
+
+	/**
+	 * Initializes the Hedera Wallet Connect Transaction Adapter.
+	 *
+	 * @param network - Optional parameter specifying the network name.
+	 * @returns A promise that resolves to the current network name.
+	 */
+	public async init(network?: NetworkName): Promise<string> {
+		const currentNetwork = network ?? this.networkService.environment;
+
+		const eventData = {
+			initData: {
+				account: this.account,
+				pairing: '',
+				topic: '',
+			},
+			wallet: SupportedWallets.HWALLETCONNECT,
+		};
+		this.eventService.emit(WalletEvents.walletInit, eventData);
+		LogService.logInfo('✅ Hedera Wallet Connect Handler Initialized');
+		return currentNetwork;
+	}
+
+	/**
+	 * Registers the Hedera WalletConnect transaction adapter.
+	 * This method registers the transaction handler and connects to WalletConnect.
+	 *
+	 * @returns A promise that resolves to an object containing the account information.
+	 */
+	public async register(): Promise<InitializationData> {
+		Injectable.registerTransactionHandler(this);
+		LogService.logTrace('Hedera WalletConnect registered as handler');
+
+		await this.connectWalletConnect();
+
+		return Promise.resolve({
+			account: this.getAccount(),
+		});
+	}
+
+	/**
+	 * Connects to the Hedera WalletConnect.
+	 *
+	 * @param network - Optional. The network to connect to. If not provided, the default network from the network service will be used.
+	 * @returns A promise that resolves to a string representing the current network.
+	 * @throws If there is an error initializing the Hedera WalletConnect or retrieving account information.
+	 */
+	public async connectWalletConnect(network?: string): Promise<string> {
+		const currentNetwork = network ?? this.networkService.environment;
+
+		try {
+			// TODO: to ENV file ⬇️
+			const projectId = '8fc26370383a50de1c3bd638d334292e';
+			const metadata: SignClientTypes.Metadata = {
+				name: 'name',
+				description: 'description',
+				url: 'https://wc.hgraph.app/',
```

Comment:
why this url?

Reply:
Good question, I have it from the old tests....